file/files.go

```diff
@@ -0,0 +1,116 @@
+package file
+
+import (
+	"bytes"
+	"compress/zlib"
+	"crypto/sha1"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"time"
+)
+
+// Init sets up a file for dotfile to track.
+// Returns the alias for the newly tracked file.
+func Init(d *Data, filePath, altName string) (string, error) {
+	var (
+		alias string
+		err   error
+	)
+
+	if _, err = os.Stat(filePath); os.IsNotExist(err) {
+		return "", fmt.Errorf("\"%#v\" not found", filePath)
+	}
+
+	// Get the full path so that it can later turn it into a relative path.
+	fullPath, err := filepath.Abs(filePath)
+	if err != nil {
+		return "", err
+	}
+
+	alias = altName
+	if altName == "" {
+		alias, err = pathToAlias(fullPath)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	if err = d.setup(); err != nil {
+		return "", err
+	}
+
+	// Replace the full path with a relative path.
+	re := regexp.MustCompile(d.Home)
+	relativePath := re.ReplaceAllString(fullPath, "~")
+
+	if err = d.save(alias, &trackedFile{
+		Path: relativePath,
+	}); err != nil {
+		return "", err
+	}
+	return alias, nil
+}
+
+// Commit hashes and saves the current state of a tracked file.
+func Commit(d *Data, alias, message string) (string, error) {
+	file, err := d.getTrackedFile(alias)
+	if err != nil {
+		return "", err
+	}
+
+	f, err := os.Open(file.getFullPath(d.Home))
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	fileBytes, err := ioutil.ReadAll(f)
+	if err != nil {
+		return "", err
+	}
+
+	hash := fmt.Sprintf("%x", sha1.Sum(fileBytes))
```

Comment:
`string(sha1.Sum(fileBytes))`?

Reply:
Not quite
`cannot convert sha1.Sum(fileBytes) (type [20]byte) to type string`